web/libs/editor/src/components/ImageView/ImageView.jsx

```diff
@@ -1281,32 +1281,50 @@ const EntireStage = observer(
 
 const ImageLayer = observer(({ item }) => {
   const imageEntity = item.currentImageEntity;
-  const image = useMemo(() => {
-    const ent = item.currentImageEntity;
+  const konvaImageRef = useRef();
+  const [loadedImage, setLoadedImage] = useState(null);
 
-    if (ent && ent.downloaded) {
+  // Load image with proper CORS and load event
+  useEffect(() => {
+    if (imageEntity?.downloaded && imageEntity.currentSrc) {
       const img = new window.Image();
-      img.src = ent.currentSrc;
-      img.width = Number.parseInt(ent.naturalWidth);
-      img.height = Number.parseInt(ent.naturalHeight);
-      return img;
+      img.crossOrigin = "anonymous";
+      img.src = `${imageEntity.currentSrc}?cb=${Date.now()}`; // avoid tainted cache
+      img.width = Number(imageEntity.naturalWidth);
```

Comment:
the original code is so confusing... `naturalWidth` and `naturalHeight` are coming straight from the image and always numbers, I don't know why did we have this transform. let's add a comment
```suggestion
      // @todo `naturalWidth` and `naturalHeight` should always be numbers; check and remove transform
      img.width = Number(imageEntity.naturalWidth);
```

Reply:
It is as you mention, we do not need this casting so I will remove it